code/_core/obj/structure/totem.dm

```diff
@@ -0,0 +1,243 @@
+	icon_state = "health"
+
+/obj/structure/totem/health_heal/totemic_effect() //copy paste from staff of healing's think, with slight modifications
+	var/turf/T = get_turf(src)
+	for(var/mob/living/L in viewers(4,T))
+		if(L.dead)
+			continue
+		if(L.loyalty_tag != affecting_faction) //!= because we want to only affect allies
+			continue
+		if(!istype(L.health))
+			continue
+		if(L.health.health_current >= L.health.health_max)
+			continue
+		if(L.health.get_loss(BRUTE))
+			L.brute_regen_buffer += (3 + (3 * leveled_effect))
+		if(L.health.get_loss(BURN))
+			L.burn_regen_buffer += (3 + (3 * leveled_effect))
+		if(L.health.get_loss(TOX))
+			L.tox_regen_buffer += (3 + (3 * leveled_effect))
+		var/obj/effect/temp/healing/H = new(L.loc,10,COLOR_RED)
+		INITIALIZE(H)
+		GENERATE(H)
+		FINALIZE(H)
+
+/obj/structure/totem/health_deal
+	name = "totem of health degeneration"
+	desc_extended = "A totem that will damage the caster's enemies' health."
+	icon_state = "healthloss"
+
+/obj/structure/totem/health_deal/totemic_effect() //copy paste from staff of healing's think, with slight modifications
+	var/turf/T = get_turf(src)
+	for(var/mob/living/L in viewers(4,T))
+		if(L.dead)
+			continue
+		if(L.loyalty_tag == affecting_faction) //== because we dont want to affect allies
+			continue
+		if(!istype(L.health))
+			continue
+		L.brute_regen_buffer -= (3 + (3 * leveled_effect))
+		L.burn_regen_buffer -= (3 + (3 * leveled_effect))
+		L.tox_regen_buffer -= (3 + (3 * leveled_effect))
+		var/obj/effect/temp/electricity/H = new(L.loc,10,COLOR_RED)
+		INITIALIZE(H)
+		GENERATE(H)
+		FINALIZE(H)
+
+/obj/structure/totem/stamina_heal
+	name = "totem of stamina regeneration"
+	desc_extended = "A totem that will restore the caster's and their allies' stamina."
+	icon_state = "stamina"
+
+/obj/structure/totem/stamina_heal/totemic_effect() //will need testing and help to balance this
+	var/turf/T = get_turf(src)
+	for(var/mob/living/L in viewers(4,T))
+		if(L.dead)
+			continue
+		if(L.loyalty_tag != affecting_faction)
+			continue
+		if(!istype(L.health))
+			continue
+		if(L.health.stamina_current >= L.health.stamina_max)
+			continue
+		if(L.health.get_stamina_loss())
+			L.stamina_regen_buffer += (3 + (3 * leveled_effect))
+		var/obj/effect/temp/healing/H = new(L.loc,10,COLOR_GREEN)
+		INITIALIZE(H)
+		GENERATE(H)
+		FINALIZE(H)
+
+/obj/structure/totem/stamina_deal
+	name = "totem of stamina degeneration"
+	desc_extended = "A totem that will damage the caster's enemies' stamina."
+	icon_state = "staminaloss"
+
+/obj/structure/totem/stamina_deal/totemic_effect() //will need testing and help to balance this
+	var/turf/T = get_turf(src)
+	for(var/mob/living/L in viewers(4,T))
+		if(L.dead)
+			continue
+		if(L.loyalty_tag == affecting_faction)
+			continue
+		if(!istype(L.health))
+			continue
+		L.stamina_regen_buffer -= (3 + (3 * leveled_effect))
+		var/obj/effect/temp/electricity/H = new(L.loc,10,COLOR_GREEN)
```

Comment:
You should use CREATE(/obj/effect/temp/healing/,get_turf(L)) instead and remove the INITIALIZE(H) and friends.